src/pepr/operator/controllers/network/policies.ts

```diff
@@ -17,6 +18,37 @@ import { defaultDenyAll } from "./defaults/default-deny-all";
 import { generate } from "./generate";
 import { allowAmbientHealthprobes } from "./generators/ambientHealthprobes";
 
+/**
+ * Finds the SSO client that matches the given service selector
+ */
+export function findMatchingClient(pkg: UDSPackage, serviceSelector: Record<string, string>) {
+  if (!serviceSelector) return undefined;
+
+  return pkg.spec?.sso?.find(
+    sso =>
+      sso.enableAuthserviceSelector &&
+      ((serviceSelector["app.kubernetes.io/name"] &&
+        sso.enableAuthserviceSelector["app.kubernetes.io/name"] ===
+          serviceSelector["app.kubernetes.io/name"]) ||
+        (serviceSelector.app && sso.enableAuthserviceSelector.app === serviceSelector.app)),
+  );
+}
+
+/**
+ * Generates a safe description for network policies
+ */
+export function getPolicyDescription(
+  port: number,
+  clientId: string | undefined,
+  gateway: string,
+  isAmbient: boolean,
+): string {
+  if (isAmbient && clientId) {
+    return `${port}-${clientId} Istio ${gateway} gateway (ambient)`;
+  }
+  return `${port}-service Istio ${gateway} gateway`;
+}
```

Comment:
This function is named very generically but seems specific to ambient authservice - might be good to rename accordingly.